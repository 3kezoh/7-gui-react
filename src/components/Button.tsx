import { cn } from "../utils";

type ButtonProps = React.ComponentPropsWithoutRef<"button">;

export function Button(props: ButtonProps) {
  const { className, disabled, ...rest } = props;

  const buttonClassName = cn(
    "border border-black px-2",
    className,
    {
      "transition-colors hover:text-white hover:bg-black": !disabled,
    },
    {
      "cursor-not-allowed bg-gray-200 text-gray-400 border-gray-400": disabled,
    },
  );

  return (
    <button
      className={buttonClassName}
      disabled={disabled}
      type="button"
      {...rest}
    />
  );
}
