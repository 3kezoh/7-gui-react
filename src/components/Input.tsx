import { cn } from "../utils";

type InputProps = React.ComponentPropsWithoutRef<"input">;

export function Input(props: InputProps) {
  const { className, name, ...rest } = props;
  const inputClassName = cn("border border-black w-full max-w-xs", className);

  return (
    <input
      className={inputClassName}
      type="text"
      id={name}
      name={name}
      {...rest}
    />
  );
}
