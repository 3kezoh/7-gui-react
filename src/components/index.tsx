export * from "./Counter";
export * from "./FlightBooker";
export * from "./TemperatureConverter";
