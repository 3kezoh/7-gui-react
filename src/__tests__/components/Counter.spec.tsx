import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Counter } from "../../components";

describe("Counter", () => {
  it("should be in the document", () => {
    render(<Counter />);

    const counter = screen.getByTestId("counter");

    expect(counter).toBeInTheDocument();
  });

  it("should render the counter with the correct initial value", () => {
    render(<Counter initialValue={10} />);

    const countElement = screen.getByText(10);

    expect(countElement).toBeInTheDocument();
  });

  it("should render the counter with the correct value after incrementing", async () => {
    const user = userEvent.setup();

    render(<Counter />);

    const incrementElement = screen.getByRole("button", { name: /count/i });

    await user.click(incrementElement);

    const countElement = screen.getByText(1);

    expect(countElement).toBeInTheDocument();
  });

  it("should render the counter with the correct value after incrementing multiple times", async () => {
    const user = userEvent.setup();

    render(<Counter />);

    const incrementElement = screen.getByRole("button", { name: /count/i });

    await user.tripleClick(incrementElement);

    const countElement = screen.getByText(3);

    expect(countElement).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<Counter />);

    expect(asFragment()).toMatchSnapshot();
  });
});
