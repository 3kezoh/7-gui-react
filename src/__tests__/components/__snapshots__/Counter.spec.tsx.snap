// Vitest Snapshot v1

exports[`Counter > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="counter"
    data-testid="counter"
  >
    <div
      class="count"
      data-testid="count"
    >
      0
    </div>
    <button
      class="button"
      data-testid="increment"
    >
      Count
    </button>
  </div>
</DocumentFragment>
`;
